### ISSUE TITLE
Disabling persistance doesn't work

### ISSUE THREAD

**Issue by MarcusWichelmann** [docker-library/redis#159]
I'm starting the redis-service using docker-compose:
```
service-redis:
    image: redis:latest
    container_name: service-redis
    restart: always
```
So according to the documentation https://hub.docker.com/_/redis/ the persistance mode should be disabled. But when I stop and start the docker-container (without removing it) the data is kept in /data/dump.rdb and restored on restart.

Have I missed any setting to completely disable this persistance or is this a bug?

Kind regards,
Marcus Wichelmann

**Comment by wglambert** [comment]
https://redis.io/topics/persistence#snapshotting
>By default Redis saves snapshots of the dataset on disk, in a binary file called dump.rdb

The docs do seem a bit misleading in that regard, as we the only thing we do is disable protected-mode after installing the .tar.gz from http://download.redis.io/

**Comment by tao12345666333** [comment]
Redis provides a different range of persistence options:

* The RDB persistence performs point-in-time snapshots of your dataset at specified intervals.
* The AOF persistence logs every write operation received by the server, that will be played again at server startup, reconstructing the original dataset. Commands are logged using the same format as the Redis protocol itself, in an append-only fashion. Redis is able to rewrite the log on background when it gets too big.
* If you wish, you can disable persistence at all, if you want your data to just exist as long as the server is running.
* It is possible to combine both AOF and RDB in the same instance. Notice that, in this case, when Redis restarts the AOF file will be used to reconstruct the original dataset since it is guaranteed to be the most complete.

Now you only used RDB, you can change `save` configuration to disable it.

**Comment by MarcusWichelmann** [comment]
Ah, okay.
Is there a way to disable snapshotting too, or do I have to create an own image based on the redis image to configure the server using a `redis.conf`?

**Comment by tao12345666333** [comment]
You can just set `save` to '' like below

```
(Tao) ➜  ~ sudo docker run --name redis2 -d redis redis-server --save ''
cc9301216dfa3012fcc6a1b78d0ae6ef691fb13e932f4b90258f02a57fd94d3b
(Tao) ➜  ~ sudo docker exec -it redis2 /bin/bash
root@cc9301216dfa:/data# redis-cli 
127.0.0.1:6379> CONFIG GET save
1) "save"
2) ""
127.0.0.1:6379> exit
root@cc9301216dfa:/data# 


```

**Comment by MarcusWichelmann** [comment]
Thank you, that did the trick!

**Comment by Fonata** [comment]
The behaviour described by @MarcusWichelmann is not related to Docker, but Redis itself: 
> data is kept in dump.rdb and restored on restart.


I found that disabling persistence in the config is not enough. It is also really important to remove the `dump.rdb` file, because otherwise data will be loaded on every Redis start. 

In my case this caused massive problems with customers because old data kept creeping into Redis. I wrote this Bash one-liner to see it happen:
````bash
printf flushing... && redis-cli flushall && redis-cli keys '*'|wc && printf "restarting redis..." && sudo service redis restart && echo "OK" && sleep 9 && echo Keys after restart: && redis-cli keys '*'|wc
````

The last command would show a large number of keys where I expected none.

My solution was to delete file that the `dir` and `dbfilename` configs point to. By default, this is "/var/lib/redis/dump.rdb".
